src/coreclr/jit/gentree.cpp

```diff
@@ -20985,22 +20985,24 @@ GenTree* Compiler::gtNewSimdBinOpNode(
 
 #if defined(TARGET_XARCH)
         case GT_RSZ:
+        case GT_LSH:
         {
             // We don't have actual instructions for shifting bytes, so we'll emulate them
             // by shifting 32-bit values and masking off the bits that should be zeroed.
 
             assert(varTypeIsByte(simdBaseType));
 
             intrinsic =
-                GenTreeHWIntrinsic::GetHWIntrinsicIdForBinOp(this, op, op1, op2ForLookup, TYP_INT, simdSize, false);
+                GenTreeHWIntrinsic::GetHWIntrinsicIdForBinOp(this, op, op1, op2ForLookup,
+                                                             op == GT_RSZ ? TYP_INT : TYP_SHORT, simdSize, false);
```

Comment:
Does this need to be `TYP_SHORT` for left shift?

`pslld` is also available for SSE2 and should achieve the same thing, correct? We typically prefer `int` where it is possible since instructions operating on `32-bit elements` tend to have better overall throughput and optimization opportunities.

Reply:
You're also right, it doesn't need to be `TYP_SHORT`. I hadn't originally considered that it could be `TYP_INT` as well.

I updated the type here to be `TYP_INT`.